### PR TITLE
Variable Groups bug fix

### LINKED ISSUE #889: SPK smoke tests failure due to variables not populated in variable group
https://dev.azure.com/epicstuff/bedrock/_build/results?buildId=9813&view=results

The front end pipeline failed downstream as a consequence. The docker image couldn’t be built.

### PR DESCRIPTION
**Pull request by sarath-p**
Closes https://github.com/microsoft/bedrock/issues/889

### CODE REVIEW

File: src/lib/pipelines/variableGroup.test.ts [hunk 1]
```diff
@@ -406,97 +405,76 @@ describe("authorizeAccessToAllPipelines", () => {
...
 
-    const variables: IVariablesMap[] = [];
-    variables.push(var1);
-
-    let map: IVariablesMap | undefined;
-    const expectedMap: any = `{"0":{"var1":{"isSecret":false,"value":"val1"}}}`;
-    map = await buildVariablesMap(variables);
+    const expectedMap: any = `{"var1":{"isSecret":false,"value":"val1"}}`;
+    const map = await buildVariablesMap(variables);
     expect(JSON.stringify(map)).toEqual(expectedMap);
```
Comment (dennisseah) [inline src/lib/pipelines/variableGroup.test.ts]: preferred not to use `any` and do object compare. Can we do this?
```
const variables: IVariableGroupDataVariable = {
  var1: {
    isSecret: false, 
    value: "val1"
  }
};
const map = await buildVariablesMap(variables);
expect(map)).toEqual(variables);
```

File: src/lib/pipelines/variableGroup.test.ts [hunk 2]
```diff
@@ -406,97 +405,76 @@ describe("authorizeAccessToAllPipelines", () => {
...
-    variables.push(var1);
-    variables.push(var2);
-
-    let secretsMap: IVariablesMap | undefined;
-    const expectedMap: any = `{"0":{"secret1":{"enabled":false}},"1":{"secret2":{"enabled":true}}}`;
-    secretsMap = await buildVariablesMap(variables);
+    const expectedMap =
+      '{"secret1":{"enabled":false},"secret2":{"enabled":true}}';
+    const secretsMap = await buildVariablesMap(variables);
     expect(JSON.stringify(secretsMap)).toEqual(expectedMap);
```
Comment (dennisseah) [inline src/lib/pipelines/variableGroup.test.ts]: same here, can we do object compare
```
expect(secretsMap).toEqual(variables);
```

File: src/lib/pipelines/variableGroup.test.ts [hunk 3]
```diff
@@ -406,97 +405,76 @@ describe("authorizeAccessToAllPipelines", () => {
...
 
   test("should create empty variable map with no secrets", async () => {
-    const variables: IVariablesMap[] = [];
-    let secretsMap: IVariablesMap | undefined;
-    const expectedMap: any = `{}`;
-    secretsMap = await buildVariablesMap(variables);
+    const variables: IVariableGroupDataVariable = {};
+    const expectedMap = `{}`;
+    const secretsMap = await buildVariablesMap(variables);
     expect(JSON.stringify(secretsMap)).toEqual(expectedMap);
```
Comment (dennisseah) [inline src/lib/pipelines/variableGroup.test.ts]:
```
expect(Object.keys(secretsMap).length).toBe(0); // empty map
```

**Review by dennisseah — COMMENTED** [review]
some suggestions

File: src/lib/pipelines/variableGroup.test.ts (hunk 1 above)
Comment (sarath-p) [reply src/lib/pipelines/variableGroup.test.ts]: Makes sense!

File: src/lib/pipelines/variableGroup.test.ts (hunk 2 above)
Comment (sarath-p) [reply src/lib/pipelines/variableGroup.test.ts]: Good idea to compare objects.

**Review by mtarng — APPROVED** [review]

File: src/lib/pipelines/variableGroup.test.ts [hunk 4]
```diff
@@ -381,11 +380,10 @@ describe("authorizeAccessToAllPipelines", () => {
     };
 
     let authorized: boolean | undefined;
-    let error: Error | undefined;
     try {
       authorized = await authorizeAccessToAllPipelines(variableGroup);
     } catch (err) {
-      error = err;
+      logger.info(err);
```
Comment (samiyaakhtar) [inline src/lib/pipelines/variableGroup.test.ts]: maybe `logger.error`?

**Review by samiyaakhtar — APPROVED** [review]

File: src/lib/pipelines/variableGroup.test.ts (hunk 4 above)
Comment (sarath-p) [reply src/lib/pipelines/variableGroup.test.ts]: Yes, good idea.

**Review by dennisseah — APPROVED** [review]

**Pull request merged**